torch_geometric/utils/convert.py

```diff
@@ -204,13 +204,10 @@ def from_networkx(
 
     G = nx.convert_node_labels_to_integers(G)
     G = G.to_directed() if not nx.is_directed(G) else G
-
-    if isinstance(G, (nx.MultiGraph, nx.MultiDiGraph)):
-        edges = list(G.edges(keys=False))
-    else:
-        edges = list(G.edges)
-
-    edge_index = torch.tensor(edges, dtype=torch.long).t().contiguous()
+    edge_index = torch.empty(G.number_of_edges(), 2, dtype=torch.long)
+    for i, e in enumerate(G.edges):
+        edge_index[i, 0], edge_index[i, 1] = e
+    edge_index = edge_index.t().contiguous()
```

Comment:
this is a for loop over edges, which can be slow if there are many edges. 
What if instead we call `del edges` after creating `edge_index` and allow the garbage collector to free up that memory if needed, this should make the OOM less likely.
```suggestion
    if isinstance(G, (nx.MultiGraph, nx.MultiDiGraph)):
        edges = list(G.edges(keys=False))
    else:
        edges = list(G.edges)

    edge_index = torch.tensor(edges, dtype=torch.long).t().contiguous()
    del edges
```